tests/kernel_performance/run_perf_case.pm

```diff
@@ -33,26 +37,36 @@ sub run_one_by_one {
     }
     assert_script_run("echo \"$case\" >> /root/qaset/list");
     assert_script_run("echo \')\' >> /root/qaset/list");
-    assert_script_run("/usr/share/qa/qaset/qaset reset");
-    assert_script_run("/usr/share/qa/qaset/run/performance-run.upload_Beijing");
-    while (1) {
-
-        #wait for case running completd with /var/log/qaset/control/DONE
-        if (script_run("ls /var/log/qaset/control/ | grep DONE") == "DONE") {
-            last;
+    for (; $i <= $repeat; $i++) {
+        assert_script_run("/usr/share/qa/qaset/qaset reset");
+        assert_script_run("/usr/share/qa/qaset/run/performance-run.upload_Beijing");
+        my $time_out = get_var("RUN_TIME_MINUTES", "180");
+        while (1) {
+            #wait for case running completd with /var/log/qaset/control/DONE
+            if (script_run("ls /var/log/qaset/control/ | grep DONE") == "DONE") {
+                last;
+            }
+            if ($time_out == 0) {
+                die "Test run didn't finish within time limit";
+            }
+            sleep 60;
+            diag "Test is not finished yet, sleeping with $time_out minutes remaining";
+            --$time_out;
+        }
+        if ($i == $repeat) {
+            power_action('poweroff');
         }
-        if ($time_out == 0) {
-            die "Test run didn't finish within time limit";
+        else {
+            power_action('reboot', textmode => 1, keepconsole => 1);
+            switch_from_ssh_to_sol_console(reset_console_flag => 'on');
+            check_screen('login_screen', 300);
```

Comment:
You are using `check_screen` without checking the return code. I suggest to just use `assert_screen` here

Reply:
Updated.